src/azure_devops_artifacts_helpers/tox_venv.py

```diff
@@ -19,14 +19,25 @@
     if tox_v4:
         from tox.plugin import impl
         from tox.tox_env.api import ToxEnv
+        from tox.config.sets import EnvConfigSet
+        from tox.session.state import State
 
         def tox_testenv_create(venv, action):
             """Empty method if tox v4 installed."""
 
         @impl
         def tox_on_install(tox_env: ToxEnv, arguments: Any, section: str, of_type: str) -> None:  # noqa: U100
             """Called before executing an installation command to install artifacts keyring."""
-            tox_env.installer.install([Requirement('artifacts-keyring')], tox_env.__class__.__name__, 'pre_deps')
+            if tox_env.conf.load('azure_devops_artifacts_helpers'):
+                tox_env.installer.install([Requirement('artifacts-keyring')], tox_env.__class__.__name__, 'pre_deps')
```

Comment:
Does this use the `PIP_INDEX_URL` if it is set? (I know it is not part of this change.) It is a general use case the `PIP_INDEX_URL` is set to point to an Azure DevOps feed, and in order to authenticate, `artifacts-keyring` is needed. I am not sure how the `install` function works, but this might be chicken-egg problem in this case.

Reply:
Good point - I think it uses cached pip packages first, but I'll have a look and check - worst case, you could probably add the index url to the pip install command in tox?

Reply:
I'll create a new issue for this specifically - I think it should be possible to directly reference the bundled wheels here on the install preventing that chicken and egg situation.